faq.md

```diff
@@ -499,6 +499,90 @@ If you are using the Flutter plugin for IntelliJ, you can launch the app in
 profile or release mode using the menu entries **Run > Flutter Run in Profile Mode**
 or **Release Mode**.
 
+### What programming paradigm does Flutter's framework use?
+
+Flutter is a multi-paradigm programming environment. Many programming techniques
+developed over the past few decades are used in Flutter, each to solve
+a particular set of problems in a clean way. In no particular order:
+
+* Composition: The primary paradigm used by Flutter is that of using small objects with narrow
+scopes of behavior, composed together to obtain more complicated effects. Most widgets
+in the Flutter widget library are built in this way.
+
+* Functional programming: Entire applications can be built with only
+[StatelessWidget](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html)s,
+which are essentially functions that describe how arguments map to other functions,
+bottoming out in primitives that compute layouts or paint graphics.
```

Comment:
If the tail of your chain of mapping functions ends in primitives that compute layouts, the state of which then affects what gets drawn, then this doesn't seem to be functional at all.

Reply:
That's like saying that Scheme isn't functional because it bottoms out to a C program that does allocation and I/O. :-)

The point here is to discuss paradigms that are used in limited parts of the system. When you deal with StatelessWidget, you are using a functional programming style. Beyond that, we use other paradigms.